packages/fetch-engine/src/getProxyAgent.ts

```diff
@@ -0,0 +1,99 @@
+import HttpsProxyAgent from 'https-proxy-agent'
+import HttpProxyAgent from 'http-proxy-agent'
+import Url from 'url'
+import { Agent as HttpsAgent } from 'https'
+import { Agent as HttpAgent } from 'http'
+
+'use strict'
+// code from https://raw.githubusercontent.com/request/request/5ba8eb44da7cd639ca21070ea9be20d611b85f66/lib/getProxyFromURI.js
+
+function formatHostname(hostname) {
+  // canonicalize the hostname, so that 'oogle.com' won't match 'google.com'
+  return hostname.replace(/^\.*/, '.').toLowerCase()
+}
+
+function parseNoProxyZone(zone) {
+  zone = zone.trim().toLowerCase()
+
+  var zoneParts = zone.split(':', 2)
```

Comment:
Let's at least turn the `var` statements into `const` if possible or `let`

Reply:
Done, thanks